utils/build-script-impl

```diff
@@ -1824,6 +1824,12 @@ function set_swiftsyntax_build_command() {
         SWIFT_TEST="$(build_directory_bin ${LOCAL_HOST} swiftpm)/swift-test"
     fi
 
+    if [ "${BUILD_LIBPARSER_ONLY}" ]; then
+        SWIFTC_USE="$(xcrun_find_tool swiftc)"
+    else
+        SWIFTC_USE="$(build_directory_bin ${LOCAL_HOST} swift)/swiftc"
+    fi
```

Comment:
I'd recommend to not conflate how swift-syntax is built with whether `-build-libparser-only` is passed or not, these are conceptually different things. I'd propose to add a new flag to indicate that swift-syntax should built with the host's swift compiler and avoid passing the explicit `--swift-build-exec`/`--swift-test-exec`/`--swiftc-exec` options.
Then on swift-syntax's build script assume that if these options are missing it should use the host's compiler.

Reply:
Actually, on second thought it may be more general and, easily applicable to other swift projects as well, if you do as you did here but also configure differently the `--swift-build-exec` and `--swift-test-exec` options.

Reply:
Done